libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -156,12 +161,22 @@ export function loadStyles(url) {
   });
 }
 
+export function isDarkMode() {
+  const { theme } = getConfig();
+  return theme === 'dark';
+}
+
 // Base styles are shared between top navigation and footer,
 // since they can be independent of each other.
 // CSS imports were not used due to duplication of file include
 export async function loadBaseStyles() {
-  const url = rootPath('base.css');
-  await loadStyles(url);
+  if (isDarkMode()) {
+    new Promise((resolve) => { loadStyle(rootPath('base.css'), resolve); })
+      .then(() => loadStyles(rootPath('dark-nav.css')));
+  } else {
+    const url = rootPath('base.css');
+    await loadStyles(url);
+  }
```

Comment:
Maybe you can define an array of Promises. Initial entry it just base. Conditionally, you can then add the dark-nav one. Then run Promise.all. In theory, this would parallelize the calls and apply the theme a bit quicker, lowering the risk of FOUC; or rather flash of unthemed content in this case 🙂 And could also speed things along a bit. This is the current timeline of these files:
<img width="530" alt="Screenshot 2024-08-20 at 17 07 31" src="https://github.com/user-attachments/assets/1a4cf910-941d-477e-8f6e-fd1baccddede">

Reply:
Since, now we're making nav `visible` in `dark-nav.css` it is made to load after `base.css`. If we load them in parallel then if `dark-nav.css` completes loading before `base.css` it will have FOUC as seen below:

<img width="1792" alt="Screenshot 2024-08-21 at 1 26 17 PM" src="https://github.com/user-attachments/assets/f912fe53-28bf-4b7e-8fc4-adda2243db9a">